test/integration/multikueue/external_job_test.go

```diff
@@ -0,0 +1,602 @@
+	"sigs.k8s.io/kueue/pkg/util/admissioncheck"
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+	testingrayjob "sigs.k8s.io/kueue/pkg/util/testingjobs/rayjob"
+	"sigs.k8s.io/kueue/pkg/webhooks"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+var _ = ginkgo.Describe("MultiKueue", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+	ginkgo.When("the external RayJob adapter is enabled", func() {
+		var (
+			managerNs *corev1.Namespace
+			worker1Ns *corev1.Namespace
+			worker2Ns *corev1.Namespace
+
+			managerMultiKueueSecret1 *corev1.Secret
+			managerMultiKueueSecret2 *corev1.Secret
+			workerCluster1           *kueue.MultiKueueCluster
+			workerCluster2           *kueue.MultiKueueCluster
+			managerMultiKueueConfig  *kueue.MultiKueueConfig
+			multiKueueAC             *kueue.AdmissionCheck
+			managerCq                *kueue.ClusterQueue
+			managerLq                *kueue.LocalQueue
+
+			worker1Cq *kueue.ClusterQueue
+			worker1Lq *kueue.LocalQueue
+
+			worker2Cq *kueue.ClusterQueue
+			worker2Lq *kueue.LocalQueue
+		)
+
+		ginkgo.BeforeAll(func() {
+			managerTestCluster.fwk.StartManager(managerTestCluster.ctx, managerTestCluster.cfg, func(ctx context.Context, mgr manager.Manager) {
+				// Set up core controllers and RayJob webhook (but not MultiKueue integration)
+				err := indexer.Setup(ctx, mgr.GetFieldIndexer())
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				cCache := schdcache.New(mgr.GetClient())
+				queues := qcache.NewManager(mgr.GetClient(), cCache)
+
+				configuration := &config.Configuration{}
+				mgr.GetScheme().Default(configuration)
+
+				failedCtrl, err := core.SetupControllers(mgr, queues, cCache, configuration)
+				gomega.Expect(err).ToNot(gomega.HaveOccurred(), "controller", failedCtrl)
+
+				failedWebhook, err := webhooks.Setup(mgr, ptr.Deref(configuration.MultiKueue.DispatcherName, config.MultiKueueDispatcherModeAllAtOnce))
+				gomega.Expect(err).ToNot(gomega.HaveOccurred(), "webhook", failedWebhook)
+
+				// Set up RayJob webhook (but not MultiKueue integration)
+				err = workloadrayjob.SetupIndexes(ctx, mgr.GetFieldIndexer())
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				rayjobReconciler := workloadrayjob.NewReconciler(
+					mgr.GetClient(),
+					mgr.GetEventRecorderFor(constants.JobControllerName))
+				err = rayjobReconciler.SetupWithManager(mgr)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				err = workloadrayjob.SetupRayJobWebhook(mgr, jobframework.WithCache(cCache), jobframework.WithQueues(queues))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				// Set up multikueue with external frameworks only
+				err = multikueue.SetupIndexer(ctx, mgr.GetFieldIndexer(), managersConfigNamespace.Name)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				cfg := &config.Configuration{}
+				mgr.GetScheme().Default(cfg)
+				cfg.MultiKueue.ExternalFrameworks = []config.MultiKueueExternalFramework{
+					{
+						Name: "RayJob.v1.ray.io",
```

Comment:
nit: Any particular reason to use RayJob rather than the batch/Job?

Reply:
Yes. We can write this test for JobSet, RayJob, etc but not Batch/Job. In Batch/Job, we remove spec.Selector and managed labels while creating object in worker.

  https://github.com/kubernetes-sigs/kueue/blob/bdbcd30e87564cefcb1bd61e4d4e09f5bafcc1c0/pkg/controller/jobs/job/job_multikueue_adapter.go#L98

https://github.com/kubernetes-sigs/kueue/blob/bdbcd30e87564cefcb1bd61e4d4e09f5bafcc1c0/pkg/controller/jobs/job/job_controller.go#L236